kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala

```diff
@@ -78,11 +78,14 @@ class JDBCMetadataStore(conf: KyuubiConf) extends MetadataStore with Logging {
 
   private def initSchema(): Unit = {
     val classLoader = Utils.getContextOrKyuubiClassLoader
+    val kyuubiShortVersion = KYUUBI_VERSION.split("-")(0)
     val initSchemaStream: Option[InputStream] = dbType match {
       case DERBY =>
-        Option(classLoader.getResourceAsStream("sql/derby/metadata-store-schema-derby.sql"))
+        Option(classLoader
+          .getResourceAsStream(s"sql/derby/metadata-store-schema-${kyuubiShortVersion}.derby.sql"))
       case MYSQL =>
-        Option(classLoader.getResourceAsStream("sql/mysql/metadata-store-schema-mysql.sql"))
+        Option(classLoader
+          .getResourceAsStream(s"sql/mysql/metadata-store-schema-${kyuubiShortVersion}.mysql.sql"))
```

Comment:
seems difficult to maintain this resource file.

we have to change the resource file name when bumping the project version.

Reply:
do you have some suggestions? @pan3793

Reply:
is it possible to use a soft link here?

Reply:
yes, we can use soft link

Reply:
we can have base file + delta file. Then base file do not need version. delta file should have version information to specify which version cause the changes.

Reply:
no we cannot use symbolic link, maven will fail.

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.2.0:resources (default-resources) on project kyuubi-server_2.12: /Users/tiliao/Documents/Code/incubator-kyuubi/kyuubi-server/target/scala-2.12/classes/sql/derby/metadata-store-schema-derby.sql -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.2.0:resources (default-resources) on project kyuubi-server_2.12: /Users/tiliao/Documents/Code/incubator-kyuubi/kyuubi-server/target/scala-2.12/classes/sql/derby/metadata-store-schema-derby.sql


Caused by: org.apache.maven.shared.filtering.MavenFilteringException: /Users/tiliao/Documents/Code/incubator-kyuubi/kyuubi-server/target/scala-2.12/classes/sql/derby/metadata-store-schema-derby.sql
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:113)
    at org.apache.maven.shared.filtering.DefaultMavenResourcesFiltering.filterResources (DefaultMavenResourcesFiltering.java:262)


https://issues.apache.org/jira/browse/MRESOURCES-269

Reply:
how do you think about @pan3793 ?

Reply:
I prefer to dynamically calculate the schema version, but the current calculation is not correct, think about 1.7.1, the expected result is 1.7.0

Reply:
Another way is listing all schema files and only always using the greatest version